### ISSUE TITLE
Device decryption blocked by power hal (/data still busy)

### ISSUE THREAD

**Issue by djselbeck** [sonyxperiadev/device-sony-common#313]
Hello,

since the treble update of the hal components the device decryption is broken. Unmounting of /data is impossible because the power hal is blocking 

`/data/misc/powerhal/rqbsvr`

I'm not yet completely familiar with the new treble system so I only created a workaround and did not open a PR. But it looks like it is necessary to restart the powerhal after decrypting /data.

My device is an Xperia XZ (tone_kagura)

Something like the diff below works for me but it is only a hack:

```diff --git a/rootdir/init.common.rc b/rootdir/init.common.rc
index fded359..c7a3fd2 100644
--- a/rootdir/init.common.rc
+++ b/rootdir/init.common.rc
@@ -51,6 +51,12 @@ on init
     write /sys/block/zram0/comp_algorithm lz4
     write /proc/sys/vm/page-cluster 0
 
+on property:vold.decrypt=trigger_reset_main
+    stop power-hal-1-0
+
+on property:vold.decrypt=trigger_post_fs_data
+    start power-hal-1-0
+
 on post-fs-data
     # Create directory used by bluetooth subsystem
     mkdir /data/misc/bluetooth 0770 bluetooth bluetooth```

**Comment by oshmoun** [comment]
could you please test if this works for you?
https://github.com/sonyxperiadev/device-sony-common/pull/316

**Comment by djselbeck** [comment]
Yes, it works correctly

**Comment by djselbeck** [comment]
Fixed with 499b64df1db61d815bbbda68179a376bd3d85e74
